### PR TITLE
fix bundle name problem #853

### PR DESCRIPTION
**Pull request by ahmedshuhel**

### CODE REVIEW

**Comment by ahmedshuhel** [comment]
@guybedford please review.

File: lib/bundle.js [hunk 1]
```diff
@@ -87,7 +89,7 @@ exports.bundle = function(moduleExpression, fileName, opts) {
 
     if (opts.inject) {
       // Add the bundle to config if the inject flag was given.
-      var bundleName = extractBundleName(fileName);
+      var bundleName = extractBundleName(fileName, systemBuilder);
```
Comment (guybedford) [inline lib/bundle.js]: Surely we can just make this `systemBuilder.getCanonicalName(fileName)`?

**Comment by guybedford** [comment]
Looks great, please let me know if my comment works out the same.

File: lib/bundle.js (hunk 1 above)
Comment (ahmedshuhel) [inline lib/bundle.js]: I believe, the func `systemBuilder.getCannonicalName(...)` needs a proper `file URL` so, if we inline this we probably have to do something like:
``` javascript
 var bundleName = builder.getCanonicalName( toFileURL(path.resolve(config.pjson.baseURL, fileName).replace(/\\/g, '/'))); 
``
This is mostly what `extractBundleName` is doing now.

Please let me know, if you prefer the above approach. I will make the change.

File: lib/bundle.js [hunk 2]
```diff
@@ -173,8 +175,9 @@ function logTree(tree) {
   ui.log('info', '');
 }
 
-function extractBundleName(fileName) {
-  return path.relative(config.pjson.baseURL, fileName.replace(/\.js$/, '')).replace(/\\/g, '/');
+function extractBundleName(fileName, builder) {
+  var file = path.resolve(config.pjson.baseURL, fileName).replace(/\\/g, '/');
```
Comment (guybedford) [inline lib/bundle.js]: The replacement of `\` with `/` is handled by toFileURL.

File: lib/bundle.js (hunk 1 above)
Comment (guybedford) [inline lib/bundle.js]: Oh right, sorry of course. That sounds fine then. I generally like inlining when something isn't used elsewhere as I find it more readable, but happy either way on this.

File: lib/bundle.js (hunk 2 above)
Comment (ahmedshuhel) [inline lib/bundle.js]: Actually, you are right here. As `toFileURL` handles `\` replacement, there's nothing much left to `extractBundleName`. Besides, `exaractBundleName` requires `builder` as a parameter, which is somewhat ugly. So, inlining makes much more sense.

Making the changes.
`      var bundleName = systemBuilder.getCanonicalName(toFileURL(path.resolve(config.pjson.baseURL, fileName)));
`

**Comment by ahmedshuhel** [comment]
Inlined `extractBundleName`. Please review.

One test is failing on Travis. I believe, nothing related to bundle/extractBundleName. 

It's trying to login to github but failing:
https://travis-ci.org/jspm/jspm-cli/jobs/70277573

```shell
․․Would you like to set up your GitHub credentials?:Yes
Enter your GitHub username:undefined
Enter your GitHub password or access token:null
Would you like to test these credentials?:Yes
Would you like to try new credentials?:Yes
Enter your GitHub username:undefined
Enter your GitHub password or access token:null
Would you like to test these credentials?:Yes
Would you like to try new credentials?:Yes
```

**Comment by guybedford** [comment]
Excellent, thanks.

**Comment by ahmedshuhel** [comment]
Thank you for your help and support.

**Pull request merged**